sdks/java/io/hcatalog/src/main/java/org/apache/beam/sdk/io/hcatalog/HCatalogIO.java

```diff
@@ -0,0 +1,507 @@
+ *       .withFilter(filterString) //optional,
+ *       should be specified if the table is partitioned
+ * }</pre>
+ *
+ *
+ *
+ * <h3>Writing using HCatalog</h3>
+ *
+ * <p>HCatalog sink supports writing of DefaultHCatRecord to a HCatalog managed source,
+ * for eg. Hive.</p>
+ *
+ * <p>To configure a HCatalog sink, you must specify a metastore URI and a table name.
+ * Other optional parameters are database, partition &amp; batchsize
+ * The destination table should exist beforehand, the transform does not create a
+ * new table if it does not exist
+ * For instance:</p>
+ *
+ * <pre>{@code
+ *
+ * Map<String, String> configProperties = new HashMap<String, String>();
+ * configProperties.put("hive.metastore.uris","thrift://metastore-host:port");
+ *
+ * pipeline
+
+ *   .apply(...)
+ *   .apply(HiveIO.write()
+ *       .withConfigProperties(configProperties) //mandatory
+ *       .withTable("employee") //mandatory
+ *       .withDatabase("default") //optional, assumes default if none specified
+ *       .withFilter(partitionValues) //optional,
+ *       should be specified if the table is partitioned
+ *       .withBatchSize(1024L)) //optional,
+ *       assumes a default batch size of 1024 if none specified
+ * }</pre>
+ *
+
+ */
+@Experimental
+public class HCatalogIO {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HCatalogIO.class);
+
+  /** Write data to Hive. */
+  public static Write write() {
+    return new AutoValue_HCatalogIO_Write.Builder().setBatchSize(1024L).build();
+  }
+
+  /** Read data from Hive. */
+  public static Read read() {
+    return new AutoValue_HCatalogIO_Read.Builder().build();
+  }
+
+  private HCatalogIO() {
+  }
+
+
+  /**
+   * A {@link PTransform} to read data using HCatalog.
+   */
+  @VisibleForTesting
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<DefaultHCatRecord>> {
+    @Nullable abstract Map<String, String> getConfigProperties();
+    @Nullable abstract String getDatabase();
+    @Nullable abstract String getTable();
+    @Nullable abstract String getFilter();
+    @Nullable abstract Coder<DefaultHCatRecord> getCoder();
+    @Nullable abstract ReaderContext getContext();
+    @Nullable abstract Integer getSplitId();
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setConfigProperties(Map<String, String> configProperties);
+      abstract Builder setDatabase(String database);
+      abstract Builder setTable(String table);
+      abstract Builder setFilter(String filter);
+      abstract Builder setCoder(Coder<DefaultHCatRecord> coder);
+      abstract Builder setSplitId(Integer splitId);
+      abstract Builder setContext(ReaderContext context);
+      abstract Read build();
+    }
+
+    /**
+     * Sets the configuration properties like metastore URI.
+     * This is mandatory
+     * @param configProperties
```

Comment:
These javadocs don't add value - it's already obvious from the method signature that there's a parameter called `configProperties`, and the `@return` tag is empty so it doesn't explain anything at all. Suggest to leave only the first two lines. Same cleanup applies to other javadoc comments.